### ISSUE TITLE
There is a bug in downloading large files in the latest version 2.6.0

### ISSUE THREAD

**Issue by Einsfat** [kingToolbox/WindTerm#2106]
The latest released version will damage the file when downloading large files. I downloaded it on two different servers and the file was damaged. Moreover, the md5 of the damaged file is consistent, so I don’t know if it is multi-threaded. Caused by fragmented download, mobaxterm download is normal

**Comment by xue-lei** [comment]
me too

**Comment by xue-lei** [comment]
All download files are damaged

**Comment by waittingsummer** [comment]
me too

**Comment by kingToolbox** [comment]
Indeed, different writing strategies are adopted for the downloading of large files and small files and files exceeding `2` megabytes are considered large files. Due to the use of `AIO+O_DIRECT+O_SYNC`, the large file needs to be manually resized to the correct size upon completion of the download, and this step is likely where the error occurred.

I will fix it as soon as possible. Also, please let me know which version has the issue, `Windows`, `macOS`, or `Linux`? Thank you.

**Comment by xue-lei** [comment]
> Indeed, different writing strategies are adopted for the downloading of large files and small files and files exceeding `2` megabytes are considered large files. Due to the use of `AIO+O_DIRECT+O_SYNC`, the large file needs to be manually resized to the correct size upon completion of the download, and this step is likely where the error occurred.
> 
> I will fix it as soon as possible. Also, please let me know which version has the issue, `Windows`, `macOS`, or `Linux`? Thank you.

windows2.6 release， 10 Prerelease no problem

**Comment by waittingsummer** [comment]
> Indeed, different writing strategies are adopted for the downloading of large files and small files and files exceeding `2` megabytes are considered large files. Due to the use of `AIO+O_DIRECT+O_SYNC`, the large file needs to be manually resized to the correct size upon completion of the download, and this step is likely where the error occurred.
> 
> I will fix it as soon as possible. Also, please let me know which version has the issue, `Windows`, `macOS`, or `Linux`? Thank you.

In fact, both Windows and Linux 2.6 release versions have this problem, and the Mac version has no test conditions.
![11111111](https://github.com/kingToolbox/WindTerm/assets/32017265/5e7cf885-53cc-4a07-aacf-7acf4dc0fa5f)
source  

![22222](https://github.com/kingToolbox/WindTerm/assets/32017265/94ccacba-b70a-47ef-b75f-ce5f7af70b46)
.tar.gz use Linux version  download

![33333333](https://github.com/kingToolbox/WindTerm/assets/32017265/fc6293f7-3382-460b-8d36-c8727f3200ed)
.tar.tgz use Windows version download

**Comment by Einsfat** [comment]
> Indeed, different writing strategies are adopted for the downloading of large files and small files and files exceeding `2` megabytes are considered large files. Due to the use of `AIO+O_DIRECT+O_SYNC`, the large file needs to be manually resized to the correct size upon completion of the download, and this step is likely where the error occurred.
> 
> I will fix it as soon as possible. Also, please let me know which version has the issue, `Windows`, `macOS`, or `Linux`? Thank you.

windows10  WindTerm_2.6.0_Windows_Portable_x86_32.zip

**Comment by BaojianXu** [comment]
> Indeed, different writing strategies are adopted for the downloading of large files and small files and files exceeding `2` megabytes are considered large files. Due to the use of `AIO+O_DIRECT+O_SYNC`, the large file needs to be manually resized to the correct size upon completion of the download, and this step is likely where the error occurred.
> 
> I will fix it as soon as possible. Also, please let me know which version has the issue, `Windows`, `macOS`, or `Linux`? Thank you.
WindTerm_2.6.0_Windows_Portable_x86_64

**Comment by BaojianXu** [comment]
> Indeed, different writing strategies are adopted for the downloading of large files and small files and files exceeding `2` megabytes are considered large files. Due to the use of `AIO+O_DIRECT+O_SYNC`, the large file needs to be manually resized to the correct size upon completion of the download, and this step is likely where the error occurred.
> 
> I will fix it as soon as possible. Also, please let me know which version has the issue, `Windows`, `macOS`, or `Linux`? Thank you.

WindTerm_2.6.0_Windows_Portable_x86_64.zip

**Comment by wtjperi2003** [comment]
me too. md5 is different

**Comment by kurniawan-yoesoef** [comment]
> Indeed, different writing strategies are adopted for the downloading of large files and small files and files exceeding `2` megabytes are considered large files. Due to the use of `AIO+O_DIRECT+O_SYNC`, the large file needs to be manually resized to the correct size upon completion of the download, and this step is likely where the error occurred.
> 
> I will fix it as soon as possible. Also, please let me know which version has the issue, `Windows`, `macOS`, or `Linux`? Thank you.

Windows 11, WindTerm_2.6.0_Windows_Portable_x86_64.zip

**Comment by jiangjianghu** [comment]
me too, Use the hex tool to open it and find that it is all data 0

**Comment by Suysker** [comment]
me too

**Comment by alexunderboots** [comment]
Same problem
Windows 10, MacOS 11.7.9
WT 2.6

**Comment by Nriver** [comment]
I've encountered hash inconsistency issues on Manjaro (Linux), which is causing significant trouble for me, particularly because I need to retrieve large backup files from the server.

As a temporary solution, I'm considering downgrading to version 2.5.0. However, I'm concerned about the integrity of my configuration file. Will the downgrade potentially corrupt it?

**Comment by zasl** [comment]
me too！！！

**Comment by ydcool** [comment]
same issue here. Windows 11

**Comment by zcteo** [comment]
+1

**Comment by xxmys** [comment]
同上，大文件下载已损坏

**Comment by Drakiat** [comment]
Same issue on 2.6.0 using macos

**Comment by xxmys** [comment]
期待修复

**Comment by swjtuhyq** [comment]
MacOs 13.6.4 2.6.0上面下载的文件损坏

**Comment by marcalath** [comment]
same here :(

Windows		11 Pro
Version		23H2
OS build		22631.3155
Processor	11th Gen Intel Core i7-1165G7 @ 2.80GHz
System type	64-bit operating system, x64-based processor

**Comment by cnzhanglu** [comment]
Does the author have a repair plan, or should they simply revert back to the old version for use?

**Comment by haleyshi** [comment]
Me too, the large files with size 0 and small files besides texts are damaged/broken. I was suspecting the transfer MODE but cannot find a way to change the transfer mode from ASCII to BINARY.

**Comment by querta** [comment]
Waiting for the fix. Same issue on Arch

**Comment by duucck** [comment]
Same issue on windows 2.6.0 release.

**Comment by lonngxiang** [comment]
same

**Comment by QQ853695231** [comment]
> > Indeed, different writing strategies are adopted for the downloading of large files and small files and files exceeding `2` megabytes are considered large files. Due to the use of `AIO+O_DIRECT+O_SYNC`, the large file needs to be manually resized to the correct size upon completion of the download, and this step is likely where the error occurred.
> > 
> > I will fix it as soon as possible. Also, please let me know which version has the issue, `Windows`, `macOS`, or `Linux`? Thank you.
> 
> Windows 11, WindTerm_2.6.0_Windows_Portable_x86_64.zip

we are same environment

**Comment by kingToolbox** [comment]
This issue has been fixed. Please download the latest version [WindTerm 2.6.0](https://github.com/kingToolbox/WindTerm/releases/tag/2.6.0) or [WindTerm 2.7.0 Prerelease 1](https://github.com/kingToolbox/WindTerm/releases/tag/2.7-prerelease). I apologize for the delay in releasing the fixed version, as I have been traveling and working without reliable internet access.
